[BUG]: Unable to deploy ZenML server on GCP with existing SQL server

### Contact Details [Optional]

_No response_

### System Information

[zenml info -a -> No such command 'info']

ZenML version: 0.33.0
Python version: 3.10.8
Platform information: {'os': 'mac', 'mac_version': '13.0.1'}
Integrations: ['gcp', 'mlflow']

### What happened?

Using `zenml deploy --config config.yaml` with `deploy_db: false` in the config file with the necessary database fields filled, I got an error about an Invalid index when reading resource : `google_sql_ssl_cert.client_cert`. I expected the deployment to use the certificates I provided instead of trying to create a new one.

To be specific, in the resource defined at https://github.com/zenml-io/zenml/blob/main/src/zenml/zen_server/deploy/terraform/recipes/gcp/sql.tf#L37, I think we should have a `count = var.deploy_db? 1 : 0` to prevent the execution of this resource creation when a DB is already provided. Same for the certificate files below it. This is the solution used for the module defined just above and for the aws and azure recipes but I don't know if it would fix the whole deployment.


### Reproduction steps

1. Deploy a SQL server on GCP, create a database and download the certificates
2. Create a config.yaml file with at least the following lines:
> ```
> name: gcp
> provider: gcp
> deploy_db: false
> ...
> ```

3. zenml deploy --config config.yaml

### Relevant log output

```shell
2023-02-20T15:53:50.018+0100 [ERROR] vertex "google_sql_ssl_cert.client_cert" error: Invalid index
2023-02-20T15:53:50.018+0100 [ERROR] vertex "google_sql_ssl_cert.client_cert" error: Invalid index
2023-02-20T15:53:50.018+0100 [ERROR] vertex "google_sql_ssl_cert.client_cert (expand)" error: Invalid index

Error: Invalid index

  on sql.tf line 37, in resource "google_sql_ssl_cert" "client_cert":
  37:   instance    = module.metadata_store[0].instance_name
    ├────────────────
    │ module.metadata_store is empty tuple
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/src/zenml/zen_server/deploy/terraform/recipes/gcp/sql.tf
+++ b/src/zenml/zen_server/deploy/terraform/recipes/gcp/sql.tf
@@ -33,20 +33,24 @@ module "metadata_store" {
 
 # create the client certificate for CloudSQL
 resource "google_sql_ssl_cert" "client_cert" {
+  count       = var.deploy_db? 1 : 0
   common_name = "sql-cert"
   instance    = module.metadata_store[0].instance_name
 }
 
 # create the certificate files
 resource "local_file" "server-ca" {
+  count    = var.deploy_db? 1 : 0
   content  = google_sql_ssl_cert.client_cert.server_ca_cert
   filename = "./server-ca.pem"
 }
 resource "local_file" "client-cert" {
+  count    = var.deploy_db? 1 : 0
   content  = google_sql_ssl_cert.client_cert.cert
   filename = "./client-cert.pem"
 }
 resource "local_file" "client-key" {
+  count    = var.deploy_db? 1 : 0
   content  = google_sql_ssl_cert.client_cert.private_key
   filename = "./client-key.pem"
 }

--- a/src/zenml/zen_server/deploy/terraform/recipes/gcp/sql.tf
+++ b/src/zenml/zen_server/deploy/terraform/recipes/gcp/sql.tf
@@ -41,16 +41,16 @@ resource "google_sql_ssl_cert" "client_cert" {
 # create the certificate files
 resource "local_file" "server-ca" {
   count    = var.deploy_db? 1 : 0
-  content  = google_sql_ssl_cert.client_cert.server_ca_cert
+  content  = google_sql_ssl_cert.client_cert[0].server_ca_cert
   filename = "./server-ca.pem"
 }
 resource "local_file" "client-cert" {
   count    = var.deploy_db? 1 : 0
-  content  = google_sql_ssl_cert.client_cert.cert
+  content  = google_sql_ssl_cert.client_cert[0].cert
   filename = "./client-cert.pem"
 }
 resource "local_file" "client-key" {
   count    = var.deploy_db? 1 : 0
-  content  = google_sql_ssl_cert.client_cert.private_key
+  content  = google_sql_ssl_cert.client_cert[0].private_key
   filename = "./client-key.pem"
 }

